### PR TITLE
Bump Python version, assorted others

### LINKED ISSUE #17105: Update to Python 3.9
## Reason
The packages we require in Conda which allows Peacock to function, have progressed to using Python 3.9. These new libraries are also required while using Peacock in MacOS Big Sur.

## Design
Bump the Python version allowed in moose-tools

## Impact
We will be keeping up to date with Conda's library stacks, as well as allowing Peacock on Big Sur to function

### PR DESCRIPTION
**Pull request by milljm**
Bump the version of latest Python allowed to 3.9.
Bump support libraries to their latest versions.

Closes #17105

### CODE REVIEW

**Review by aeslaughter — DISMISSED** [review]

**Review by aeslaughter — DISMISSED** [review]

**Review by aeslaughter — DISMISSED** [review]

**Review by aeslaughter — DISMISSED** [review]

**Comment by milljm** [comment]
@aeslaughter Seems to be VTK version issues causing the image diffs. I am looking into it. But unfortunately the way it stands right now, the only version of VTK available with Python 3.9, is 9.0.1.

**Comment by aeslaughter** [comment]
I am alright with moving up to VTK9, do you need me to look into it?

**Comment by milljm** [comment]
I think VTK 9 will work, so long as we re-gold things. But, there are other issues (as seen in the Mac Test failures).

**Review by aeslaughter — DISMISSED** [review]

**Comment by milljm** [comment]
I know there is a libMesh update in progress @lindsayad. These changes will certainly need to be included in your PR: https://github.com/idaholab/moose/pull/18268

Which ever goes in first I suppose?

**Comment by lindsayad** [comment]
sounds good. I suspect that your PR here will go in first, and then I'll rebase

**Comment by milljm** [comment]
I have another commit that I want to include with this conda package update. Stand by.

**Comment by milljm** [comment]
@lindsayad This is a hefty change. Let me know if you help updating libMesh.

I've basically consolidated the conda_build_config.yaml file, so stop forgetting to update multiple packages when necessary. And for sanity purposes of having it all in one place.

File: conda/mpich/conda_build_config.yaml [hunk 1]
```diff
@@ -1,15 +1,26 @@
-CONDA_BUILD_SYSROOT:           # [osx]
-  - /opt/MacOSX10.9.sdk        # [osx]
+#### The idea here is that anything below something else requires
+#### anything above that to be updated.
```
Comment (milljm) [inline conda/mpich/conda_build_config.yaml:2]: dumbest comment ever. I will change this...

**Comment by lindsayad** [comment]
> This is a hefty change. Let me know if you need help updating libMesh.

will do!

**Review by loganharbour — APPROVED** [review]

**Pull request merged**